src/main/java/com/develhope/spring/order/service/OrderService.java

```diff
@@ -41,4 +42,15 @@ public void deleteOrderById(long id) {
     public void deleteAllOrders() {
         orderRepository.deleteAll();
     }
+
+    public void updateOrderStatus(long orderId) {
```

Comment:
Considerando come stavamo discutendo che potrebbero entrare in campo, in futuro altri Status e anche per evitare di incappare in errori io prenderei lo status come parametro.
Poi se è uguale allo stato attuale semplicemente non si fa cambiarlo alla repo ma si manda solo l'ok.
Se non è uguale si cambia con flush